tests/unit/test_query.py

```diff
@@ -0,0 +1,313 @@
+                "execute_chunks_in_parallel": True,
+                "chunk_size": 1000,
+                "retries_per_chunk": 3,
+            }
+        ),
+    )
+
+
+def test_query_batch_as_query_with_apoc_iterate_false():
+    batch = QueryBatch("CREATE (n:Node {id: $id})", [{"id": 1}, {"id": 2}, {"id": 3}])
+
+    result = batch.as_query(apoc_iterate=False)
+
+    assert_that(result.query_statement, equal_to(NON_APOCH_COMMIT_QUERY))
+    assert_that(result.is_apoc, equal_to(False))
+    assert_that(
+        result.parameters,
+        has_entries(
+            {
+                "iterate_params": {
+                    "batched_parameter_sets": [{"id": 1}, {"id": 2}, {"id": 3}]
+                },
+                "batched_query": "CREATE (n:Node {id: $id})",
+                "iterable_query": UNWIND_QUERY,
+                "execute_chunks_in_parallel": True,
+                "chunk_size": 1000,
+                "retries_per_chunk": 3,
+            }
+        ),
+    )
+
+
+def test_query_batch_as_query_with_custom_parameters():
+    batch = QueryBatch("CREATE (n:Node {id: $id})", [{"id": 1}])
+
+    result = batch.as_query(
+        apoc_iterate=True,
+        chunk_size=2000,
+        execute_chunks_in_parallel=False,
+        retries_per_chunk=10,
+    )
+
+    assert_that(
+        result.parameters,
+        has_entries(
+            {
+                "execute_chunks_in_parallel": False,
+                "chunk_size": 2000,
+                "retries_per_chunk": 10,
+            }
+        ),
+    )
+
+
+def test_apoc_batch_query_response_fields():
+    expected_fields = [
+        "batches",
+        "total",
+        "timeTaken",
+        "committedOperations",
+        "failedOperations",
+        "failedBatches",
+        "retries",
+        "errorMessages",
+        "wasTerminated",
+        "updateStatistics",
+    ]
+    assert_that(set(APOC_BATCH_QUERY_RESPONSE_FIELDS), equal_to(set(expected_fields)))
+
+
+def test_unwind_query():
+    assert_that(
+        UNWIND_QUERY, equal_to("UNWIND $batched_parameter_sets as params RETURN params")
+    )
+
+
+def test_yield_clause_contains_all_fields():
+    expected_yield = f"YIELD {', '.join(APOC_BATCH_QUERY_RESPONSE_FIELDS)}"
+    assert_that(YIELD_CLAUSE, equal_to(expected_yield))
+
+
+def test_return_clause_contains_all_fields():
+    expected_return = f"RETURN {', '.join(APOC_BATCH_QUERY_RESPONSE_FIELDS)}"
+    assert_that(RETURN_CLAUSE, equal_to(expected_return))
+
+
+def test_commit_query_structure():
+    assert_that("CALL apoc.periodic.iterate" in COMMIT_QUERY, equal_to(True))
+    assert_that("YIELD" in COMMIT_QUERY, equal_to(True))
+    assert_that("RETURN" in COMMIT_QUERY, equal_to(True))
+
+
+def test_non_apoc_commit_query_structure():
+    assert_that("UNWIND" in NON_APOCH_COMMIT_QUERY, equal_to(True))
+    assert_that("CALL apoc.cypher.doIt" in NON_APOCH_COMMIT_QUERY, equal_to(True))
+    assert_that("RETURN" in NON_APOCH_COMMIT_QUERY, equal_to(True))
```

Comment:
Why are we verifying constants? Do they change at load time?

Also, it might be more clear to use something like:
```suggestion
    assert_that(
        NON_APOCH_COMMIT_QUERY,
        hamcrest.all_of(
            hamcrest.contains_string("UNWIND"),
            hamcrest.contains_string("CALL apoc.cypher.doIt"),
            hamcrest.contains_string("RETURN"),
        ),
    )
```
I dislike `equal_to(True)`

Another option is to use the pytest way:
```suggestion
    assert "UNWIND" in NON_APOCH_COMMIT_QUERY
    assert "CALL apoc.cypher.doIt" in NON_APOCH_COMMIT_QUERY
    assert "RETURN" in NON_APOCH_COMMIT_QUERY
```